src/prefect/client/schemas/actions.py

```diff
@@ -212,7 +213,7 @@ def convert_to_strings(
     ) -> Union[str, list[str]]:
         return convert_to_strings(values)
 
-    name: str = Field(..., description="The name of the deployment.")
+    name: NameOrEmpty = Field(..., description="The name of the deployment.")
```

Comment:
fail client-side on invalid deployment names instead of throwing 500 server-side